File: customview/src/main/java/com/example/customview/activity/AnimationBtnActivity.java
package com.example.customview.activity;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.Toast;

import com.example.customview.R;
import com.example.customview.widget.AnimationButton;
/**
 * 用于展示 Animation Btn 功能的 Activity。
 */

public class AnimationBtnActivity extends AppCompatActivity {

    private AnimationButton animationButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_animation_btn);
        animationButton = findViewById(R.id.animation_btn);
        animationButton.setAnimationButtonListener(new AnimationButton.AnimationButtonListener() {
            @Override
            public void onClickListener() {
                animationButton.start();
            }

            @Override
            public void animationFinish() {
                Toast.makeText(AnimationBtnActivity.this, "动画执行完毕", Toast.LENGTH_SHORT).show();
                animationButton.reset();
            }
        });
    }
}

File: customview/src/main/java/com/example/customview/activity/AnimationBtnActivity.kt
package com.example.customview.activity

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.customview.databinding.ActivityAnimationBtnBinding
import com.example.customview.widget.AnimationButton

/**
 * 用于展示 Animation Btn 功能的 Activity。
 */
open class AnimationBtnActivity : AppCompatActivity() {
    private lateinit var binding: ActivityAnimationBtnBinding
    private lateinit var animationButton: AnimationButton

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAnimationBtnBinding.inflate(layoutInflater)
        setContentView(binding.root)
        animationButton = binding.animationBtn
        animationButton.setAnimationButtonListener(object : AnimationButton.AnimationButtonListener {
            override fun onClickListener() {
                animationButton.start()
            }

            override fun animationFinish() {
                Toast.makeText(this@AnimationBtnActivity, "动画执行完毕", Toast.LENGTH_SHORT).show()
                animationButton.reset()
            }
        })
    }
}
